packages/editor/src/components/rich-text/patterns.js

```diff
@@ -20,8 +18,7 @@ export function getPatterns( { onReplace, valueToFormat } ) {
 				return record;
 			}
 
-			const start = getSelectionStart( record );
-			const text = getTextContent( record );
+			const { start, text } = record;
```

Comment:
Based on this change, is there a reason to keep `getTextContent` around?

Reply:
Unfortunately, it has been documented.

Reply:
> Unfortunately, it has been documented.

I later edited my comment with a secondary question: If it's to remain, should we be using it? Or are we fine to just let it linger as dead code for compatibility's sake?

Reply:
We can update the code, then add some more inline documentation to suggest just accessing `text`.

Reply:
Decided to leave it for now and address in a separate PR. `getTextContent` should actually stay, but it's implemented poorly: it should also filter out invisible replacement characters. In some cases, we are using `getTextContent` where we shouldn't.